### ISSUE TITLE
skip matching rules in code within functions that are only called from library functions

### ISSUE THREAD

**Issue by sara-rn** [mandiant/capa#989]
### Summary

Some functions that are not identified as library functions but are only called from library functions do match capa rules. An enhancement would be to skip matching rules associated to capabilities not only within library functions but also within functions only called from library functions.

### Motivation

As capa is aimed at detecting capabilities and behaviours within the programmer's code and not within library code this will reduce the hits an analyst have to look at that are relevant.

### Describe alternatives you've considered

Skip finding capabilities and behaviours within functions that are not originally detected as library code but it i safe to assume they are library functions since they are only called from library code.

## Additional context
Seen in a DLL where the library function __IsNonwritableInCurrentImage would call a function, not detected as library function by IDA Pro.

<img width="444" alt="Screenshot 2022-04-11 at 12 19 22" src="https://user-images.githubusercontent.com/103417144/162720601-55801299-1f0e-45d3-b7c8-9f554987704f.png">

<img width="509" alt="Screenshot 2022-04-11 at 12 19 30" src="https://user-images.githubusercontent.com/103417144/162720635-ca2e8126-d20f-4054-80cd-384773c9ca23.png">

**Comment by williballenthin** [comment]
I like this idea and its definitely in the spirit of what we've been doing with capa analysis.

Implementation-wise, we already extract enough information to generate a call graph that we could use to filter the results. TBD on whether this filtering happens after feature extraction, though, which wouldn't reduce runtime but would still reduce FP output.

**Comment by mr-tz** [comment]
Would it be possible to add this to `viv_utils` so all dependencies just get this feature? Not sure this would work there though as I think it may focus on single functions only.

**Comment by williballenthin** [comment]
we could put into viv-utils the logic that decides which functions call which functions and therefore should/nt be included in analysis (i think this is called a "dominator" in graph theory). then capa can be a client of this logic, along with other dependencies. floss might also make use of the logic.

im not sure how we'd decide "is a library function?". maybe clients bring their own list of functions that it thinks are libraries (such as via FLIRT, etc).

**Comment by mr-tz** [comment]
That sounds good. Having this available in all dependencies would be neat.

**Comment by s-ff** [comment]
I agree with implementing this in `viv-utils`, so that it can be used by others.
Should I open a PR there, or let's stick with capa for now? :)

**Comment by mr-tz** [comment]
Yes, please, `viv-utils` is good. I see potential for also using this feature in FLOSS, for example.
